iDiff/undocker.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+import argparse
+import errno
+import json
+import logging
+import os
+import tarfile
+import shutil
+
+
+LOG = logging.getLogger(__name__)
+
+
+def parse_args():
+    p = argparse.ArgumentParser()
+    p.add_argument('--tar', '-t',
+                   default='',
+                   help='The tar path to undocker')
+    p.add_argument('--ignore-errors', '-i',
+                   action='store_true',
+                   help='Ignore OS errors when extracting files')
+    p.add_argument('--output', '-o',
+                   default='.',
+                   help='Output directory (defaults to ".")')
+    p.add_argument('--verbose', '-v',
+                   action='store_const',
+                   const=logging.INFO,
+                   dest='loglevel')
+    p.add_argument('--debug', '-d',
+                   action='store_const',
+                   const=logging.DEBUG,
+                   dest='loglevel')
+    p.add_argument('--layers',
+                   action='store_true',
+                   help='List layers in an image')
+    p.add_argument('--list', '--ls',
+                   action='store_true',
+                   help='List images/tags contained in archive')
+    p.add_argument('--layer', '-l',
+                   action='append',
+                   help='Extract only the specified layer')
+    p.add_argument('--no-whiteouts', '-W',
+                   action='store_true',
+                   help='Do not process whiteout (.wh.*) files')
+    p.add_argument('image', nargs='?')
+
+    p.set_defaults(level=logging.WARN)
+    return p.parse_args()
+
+
+def main():
+    args = parse_args()
+    logging.basicConfig(level=args.loglevel)
+    # open tar file
+    img = tarfile.open(args.tar)
+    print args.tar
+    # extract manifest file from tar to read layers
+    repos = img.extractfile('manifest.json')
```

Comment:
This python script will be replaced in my next PR with go code when the docker dependency is removed.  In this current state of code, unless a .tar file without a manifest is passed, the manifest file is guaranteed to exist.  I did add a check though just in case.